### DISCUSSION TITLE
Hi,when I recompile the code,'error: command 'D:\\python\\Scripts\\cmake.exe' failed with exit code 1' always appears at the end of the cmd command window.And how to slove it?

### DISCUSSION THREAD

**Discussion by IlllIIIlllIl** [SynxFlow/SynxFlow#21] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/51728e08-25d8-4cba-9fb1-a9b54bc4c470)

**Comment by xiaxilin** [comment]
What was the full error message?

**Reply by IlllIIIlllIl** [reply]
PS C:\Users\Administrator\Desktop\SynxFlow_old> python setup.py install
running install
running bdist_egg
running egg_info
writing synxflow.egg-info\PKG-INFO
writing dependency_links to synxflow.egg-info\dependency_links.txt
writing requirements to synxflow.egg-info\requires.txt
writing top-level names to synxflow.egg-info\top_level.txt
reading manifest file 'synxflow.egg-info\SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'README.rst'
warning: no files found matching '*.asc' under directory 'synxflow'
adding license file 'LICENSE.txt'
adding license file 'authors.md'
writing manifest file 'synxflow.egg-info\SOURCES.txt'
installing library code to build\bdist.win-amd64\egg
running install_lib
running build_py
running build_ext
The cuda toolkit root path is
cmake C:\Users\Administrator\Desktop\SynxFlow_old -DCMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE=C:\Users\Administrator\Desktop\SynxFlow_old\build\lib.win-amd64-3.10\synxflow -A x64
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
CMake Warning (dev) at synxflow/CMakeLists.txt:1 (find_package):
  Policy CMP0146 is not set: The FindCUDA module is removed.  Run "cmake
  --help-policy CMP0146" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CXX  Flags: /DWIN32 /D_WINDOWS /W3 /GR /EHsc/Zi
NVCC Flags: -arch=sm_50--expt-extended-lambda
CMake Warning (dev) at synxflow/CMakeLists.txt:20 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at synxflow/CMakeLists.txt:21 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at D:/python/Lib/site-packages/cmake/data/share/cmake-3.30/Modules/FetchContent.cmake:1953 (message):
  Calling FetchContent_Populate(pybind11_sources) is deprecated, call
  FetchContent_MakeAvailable(pybind11_sources) instead.  Policy CMP0169 can
  be set to OLD to allow FetchContent_Populate(pybind11_sources) to be called
  directly for now, but the ability to call it with declared details will be
  removed completely in a future version.
Call Stack (most recent call first):
  synxflow/CMakeLists.txt:34 (FetchContent_Populate)
This warning is for project developers.  Use -Wno-dev to suppress it.

閫傜敤浜?.NET Framework MSBuild 鐗堟湰 17.11.2+c078802d4

  Performing update step for 'pybind11_sources-populate'
  -- Fetching latest from the remote origin
CUSTOMBUILD : error : RPC failed; curl 28 Recv failure: Connection was reset [C:\Users\Administrator\Desktop\SynxFlow_old\build\temp.win-amd64-3.10\Release\_deps\pybind11_sources-subbuild\pybind11_sources-populate.vcxproj]
  fatal: expected flush after ref listing
  CMake Error at C:/Users/Administrator/Desktop/SynxFlow_old/build/temp.win-amd64-3.10/Release/_deps/pybind11_sources-subbuild/pybind11_sources-populate-prefix/tmp/pybind11_sources-populate-gitupdate.cmake:17 (execute_process):
    execute_process last command failed
  Call Stack (most recent call first):
    C:/Users/Administrator/Desktop/SynxFlow_old/build/temp.win-amd64-3.10/Release/_deps/pybind11_sources-subbuild/pybind11_sources-populate-prefix/tmp/pybind11_sources-populate-gitupdate.cmake:89 (do_fetch)


C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: 鈥淐:\Users\Administrator\Desktop\SynxFlow_old\build\temp.win-amd64-3.10\Release\_deps\pybind11_sources-subbuild\CMakeFiles\274b00b177efd7214586f3993f9e4dfb\pybind11_sources-populate-update.rule;C:\Users\Administrator\Desktop\SynxFlow_old\build\temp.win-amd64-3.10\Release\_deps\pybind11_sources-subbuild\CMakeFiles\274b00b177efd7214586f3993f9e4dfb\pybind11_sources-populate-patch.rule;C:\Users\Administrator\Desktop\SynxFlow_old\build\temp.win-amd64-3.10\Release\_deps\pybind11_sources-subbuild\CMakeFiles\274b00b177efd7214586f3993f9e4dfb\pybind11_sources-populate-configure.rule;C:\Users\Administrator\Desktop\SynxFlow_old\build\temp.win-amd64-3.10\Release\_deps\pybind11_sources-subbuild\CMakeFiles\274b00b177efd7214586f3993f9e4dfb\pybind11_sources-populate-build.rule;C:\Users\Administrator\Desktop\SynxFlow_old\build\temp.win-amd64-3.10\Release\_deps\pybind11_sources-subbuild\CMakeFiles\274b00b177efd7214586f3993f9e4dfb\pybind11_sources-populate-install.rule;C:\Users\Administrator\Desktop\SynxFlow_old\build\temp.win-amd64-3.10\Release\_deps\pybind11_sources-subbuild\CMakeFiles\274b00b177efd7214586f3993f9e4dfb\pybind11_sources-populate-test.rule;C:\Users\Administrator\Desktop\SynxFlow_old\build\temp.win-amd64-3.10\Release\_deps\pybind11_sources-subbuild\CMakeFiles\ca62eeef8b12831c07c49d8df40326d3\pybind11_sources-populate-complete.rule;C:\Users\Administrator\Desktop\SynxFlow_old\build\temp.win-amd64-3.10\Release\_deps\pybind11_sources-subbuild\CMakeFiles\3ab706ebe8af59169c37883e7daeecee\pybind11_sources-populate.rule鈥濈殑鑷畾涔夌敓鎴愬凡閫€鍑猴紝浠ｇ爜涓?1銆?[C:\Users\Administrator\Desktop\SynxFlow_old\build\temp.win-amd64-3.10\Release\_deps\pybind11_sources-subbuild\pybind11_sources-populate.vcxproj]

CMake Error at D:/python/Lib/site-packages/cmake/data/share/cmake-3.30/Modules/FetchContent.cmake:1918 (message):
  Build step for pybind11_sources failed: 1
Call Stack (most recent call first):
  D:/python/Lib/site-packages/cmake/data/share/cmake-3.30/Modules/FetchContent.cmake:1609 (__FetchContent_populateSubbuild)
  D:/python/Lib/site-packages/cmake/data/share/cmake-3.30/Modules/FetchContent.cmake:2145:EVAL:2 (__FetchContent_doPopulation)
  D:/python/Lib/site-packages/cmake/data/share/cmake-3.30/Modules/FetchContent.cmake:2145 (cmake_language)
  D:/python/Lib/site-packages/cmake/data/share/cmake-3.30/Modules/FetchContent.cmake:1978:EVAL:1 (__FetchContent_Populate)
  D:/python/Lib/site-packages/cmake/data/share/cmake-3.30/Modules/FetchContent.cmake:1978 (cmake_language)
  synxflow/CMakeLists.txt:34 (FetchContent_Populate)


-- Configuring incomplete, errors occurred!
error: command 'D:\\python\\Scripts\\cmake.exe' failed with exit code 1
its the full message of the error,could you please help me see what's wrong?

**Reply by IlllIIIlllIl** [reply]
and a new issue is there:
 cuda_adaptive_time_control.cu
  tmpxft_00007728_00000000-10_cuda_adaptive_time_control.cudafe1.cpp
  Building NVCC (Device) object synxflow/lib/CMakeFiles/cuda_lib.dir/src/utilities/Release/cuda_lib_generated_cuda_devi
  ce_query.cu.obj
  cuda_device_query.cu
  cuda_device_query.cu
  tmpxft_00006e68_00000000-10_cuda_device_query.cudafe1.cpp
  Building NVCC intermediate link file synxflow/lib/CMakeFiles/cuda_lib.dir/Release/cuda_lib_intermediate_link.obj
  cuda_lib_generated_cuda_gauges_writer.cu.obj
  cuda_lib_generated_cuda_gisascii_writer.cu.obj
  cuda_lib_generated_cuda_mesh_fv.cu.obj
  cuda_lib_generated_cuda_data_bank.cu.obj
  cuda_lib_generated_cuda_advection_NSWEs.cu.obj
  cuda_lib_generated_cuda_erosion_deposition.cu.obj
  cuda_lib_generated_cuda_friction.cu.obj
  cuda_lib_generated_cuda_gradient.cu.obj
  cuda_lib_generated_cuda_hessian.cu.obj
  cuda_lib_generated_cuda_infiltration.cu.obj
  cuda_lib_generated_cuda_limiter.cu.obj
  cuda_lib_generated_cuda_melt.cu.obj
  cuda_lib_generated_cuda_transport_NSWEs.cu.obj
  cuda_lib_generated_cuda_adaptive_time_control.cu.obj
  cuda_lib_generated_cuda_device_query.cu.obj
  cuda_lib.vcxproj -> C:\Users\Administrator\Desktop\SynxFlow-master\build\temp.win-amd64-3.10\Release\synxflow\lib\Rel
  ease\cuda_lib.lib
error: command 'D:\\python\\Scripts\\cmake.exe' failed with exit code 1
how to slove it?

**Comment by xiaxilin** [comment]
It seems it is not pulling Pybind source code successfully, can you check whether you can access the Pybind 11 source code repository?

**Reply by IlllIIIlllIl** [reply]
![image](https://github.com/user-attachments/assets/a78c98b1-8dac-4332-b397-8c7fa036ff21)
thank for your help.And I can access the source code in github.My thoughts are as follows:
"FetchContent_Declare(
pybind11_sources
GIT_REPOSITORY https://github.com/pybind/pybind11.git
GIT_TAG v2.5)
Would it be more advisable to replace the above-mentioned process of fetching the github source library content in the synxflow/CMakeLists.txt file with accessing the local pybind11? If so, how do you reckon I can implement the invocation of pybind11 in the local python library?

**Reply by xiaxilin** [reply]
Yes, I think it is a good idea to try. I haven't done this before but it is certainly possible.

**Reply by xiaxilin** [reply, marked as answer]
You can use another git mirror site, for example

https://kkgithub.com/pybind/pybind11.git

This was working for me.
More information can be found at
https://github.com/runningcheese/MirrorSite?tab=readme-ov-file
